src/context.rs

```diff
@@ -26,12 +26,12 @@ use ffi::enums::{Status, Antialias, LineCap, LineJoin, FillRule};
 use ::patterns::{wrap_pattern, Pattern};
 use surface::Surface;
 
-pub struct RectangleVec {
+pub struct RectangleVec<'a> {
     ptr: *mut cairo_rectangle_list_t,
-    pub rectangles: CVec<Rectangle>,
+    pub rectangles: &'a[Rectangle], // Vec<Rectangle>,
 }
```

Comment:
https://doc.rust-lang.org/book/lifetimes.html#lifetime-elision
By way, @daym, it solves your problem if @GuillaumeGomez publish `c_vec` with upgraded `lib_c` dependence?